ui/src/app/shared/components/array-input/array-input.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as ReactForm from 'react-form';
+
+class Item {
+    public name: string;
+    public value: string;
+}
+
+class Props {
+    public setValue: (value: Item[]) => void;
+    public getValue: () => Item[];
+}
+
+class State {
```

Comment:
either (a) replace `State` with `Item`, or (b) have a field in `State` called `newItem`